### DISCUSSION TITLE
Pass Link between Capture steps

### DISCUSSION THREAD

**Discussion by rapatel0** [chhoumann/quickadd#217] [Q&A]
Hey I'm trying to pass a {{LINKCURRENT}} between two separate capture steps. 

- Capture step 1 - Creates a file with information in file A
- capture step 2 - adds info to a different file B

I would like to Pass the Link to file A in the capture formatting of the text appended to file B

Any ideas?

**Comment by chhoumann** [comment]
Coming back to this very late - sorry about that.

The trick is to wrap both captures in a Macro choice - every command in a macro shares one variable map, so a named `{{VALUE:x}}` prompted in one step is already resolved in the next.

- **Capture 1**: set *Capture To* to something like `Notes/{{VALUE:noteName}}.md` (with *Create file if it doesn't exist* enabled). You get prompted for `noteName` once, and file A is created.
- **Capture 2**: target file B, and put `[[{{VALUE:noteName}}]]` in the capture format. The variable was set in step 1, so it resolves to the link with no second prompt.

If you need more than the name (full path, an alias), add a small user script between the two captures that builds it - `params.variables.linkToA = ...` - and reference `{{VALUE:linkToA}}` in capture 2.

As for why `{{LINKCURRENT}}` didn't work: it resolves to the currently *active* file, not to files created mid-macro, so it isn't a reliable way to reference file A.

Docs: https://quickadd.obsidian.guide/docs/Choices/MacroChoice

Closing as resolved.
